modules/account-lib/src/coin/dot/transactionBuilder.ts

```diff
@@ -1,57 +1,506 @@
+import { BaseTransactionSchema, SignedTransactionSchema, SigningPayloadTransactionSchema } from './txnSchema';
 import { Transaction } from './transaction';
+import { KeyPair } from './keyPair';
+import { CreateBaseTxInfo, sequenceId, specNameType, TxMethod, Validity } from './iface';
+import Utils from './utils';
+import { AddressValidationError } from './errors';
 
 export abstract class TransactionBuilder extends BaseTransactionBuilder {
-  private _transaction: Transaction;
+  protected _transaction: Transaction;
+  protected _keyPair: KeyPair;
+  protected _sender: string;
+
+  protected _blockNumber: number;
+  protected _blockHash: string;
+  protected _genesisHash: string;
+  protected _metadataRpc: string;
+  protected _specVersion: number;
+  protected _transactionVersion: number;
+  protected _specName: specNameType;
+  protected _chainName: string;
+  protected _nonce: number;
+  protected _tip?: number;
+  protected _eraPeriod?: number;
+  protected _registry: TypeRegistry;
+  protected _method?: TxMethod;
 
   constructor(_coinConfig: Readonly<CoinConfig>) {
     super(_coinConfig);
+    this._transaction = new Transaction(_coinConfig);
+    this.staticsConfig();
+  }
+  /**
+   *
+   * Sets the address of sending account.
+   *
+   * @param {BaseAddress} address The SS58-encoded address.
+   * @returns {TransactionBuilder} This transaction builder.
+   *
+   * @see https://wiki.polkadot.network/docs/build-transaction-construction
+   */
+  sender({ address }: BaseAddress): this {
+    this.validateAddress({ address });
+    this._sender = address;
+    this._transaction.sender(address);
+    return this;
+  }
+  /**
+   *
+   * The number of blocks after the checkpoint for which a transaction is valid. If zero, the transaction is immortal.
+   *
+   * @param {number | undefined} maxDuration
+   * @returns {TransactionBuilder} This transaction builder.
+   *
+   * @see https://wiki.polkadot.network/docs/build-transaction-construction
+   */
+  durationConfig({ maxDuration }: { maxDuration: number | undefined }): this {
+    if (maxDuration !== undefined) {
+      this.validateValue(new BigNumber(maxDuration));
+      this._eraPeriod = maxDuration;
+    }
+    return this;
+  }
+  /**
+   *
+   * The nonce for this transaction.
+   *
+   * @param {number} nonce
+   * @returns {TransactionBuilder} This transaction builder.
+   *
+   * @see https://wiki.polkadot.network/docs/build-transaction-construction
+   */
+  sequenceId(nonce: sequenceId): this {
+    const value = new BigNumber(nonce.value);
+    this.validateValue(value);
+    this._nonce = value.toNumber();
+    return this;
+  }
+  /**
+   *
+   * The tip to increase transaction priority.
+   *
+   * @param {number | undefined} tip optional
+   * @returns {TransactionBuilder} This transaction builder.
+   *
+   * @see https://wiki.polkadot.network/docs/build-transaction-construction
+   */
+  tip(tip: number | undefined): this {
```

Comment:
This is a fee of type tip: https://docs.google.com/document/d/1cMGyXa8GwgwDMcZ2Uv7dMDsqRWlIEp1MRdVgpPW8p1o/edit#bookmark=id.nw0j8jgi5fc5